### PR TITLE
Grammar: Add SET sql_log_bin statement to MySQL dialect

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

Pretty simple, just need to add a "ON|OFF"/"TRUE|FALSE" keyword option to the existing `SET` statement to parse correctly. i.e. "ON|OFF"/"TRUE|FALSE" should match keyword capitalisation not variable capitalisation.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 1]
```diff
@@ -316,6 +316,14 @@ class TableConstraintSegment(BaseSegment):
         name="declared_variable",
         type="variable",
     ),
+    BooleanDynamicSystemVariablesGrammar=OneOf(
+        # Boolean dynamic system varaiables can be set to ON/OFF, TRUE/FALSE, or 0/1:
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_mysql.py:320]: 0/1 is not covered here. Guessing not needed in this Grammar but should we at least add a test case to show this works anyway?

Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_mysql.py:320]: yeah this grammar is just to get these parsing as keywords and therefore have the correct capitalisation. Best to let 0/1 parse in the same way as the rest of the integers 👍 

I've added an extra unit test

**Review by tunetheweb — APPROVED** [review]
LGTM.

Might want to update that first comment and AND and OR as doesn’t seem to match what you ended up implementing?

**Comment by jpy-git** [comment]
> Might want to update that first comment about AND and OR as doesn’t seem to match what you ended up implementing?

Ah yep that was a typo in the PR description sorry, fixed it now 👍

**Pull request merged**